pkg/devfile/parser/context/context.go

```diff
@@ -90,11 +104,24 @@ func (d *DevfileCtx) Populate() (err error) {
 
 // PopulateFromURL fills the DevfileCtx struct with relevant context info
 func (d *DevfileCtx) PopulateFromURL() (err error) {
-
-	_, err = url.ParseRequestURI(d.url)
+	u, err := url.ParseRequestURI(d.url)
 	if err != nil {
 		return err
 	}
+	if !strings.HasSuffix(d.url, ".yaml") {
+		u.Path = path.Join(u.Path, "devfile.yaml")
+		param := util.HTTPRequestParams{
+			URL: u.String(),
+		}
+		if _, err = util.HTTPGetRequest(param, 0); err != nil {
+			u.Path = path.Join(u.Path, ".devfile.yaml")
+			param.URL = u.String()
+			if _, err = util.HTTPGetRequest(param, 0); err != nil {
+				return fmt.Errorf("the provided url is not a valid yaml filepath, and no devfile.yaml or .devfile.yaml under provided url: %s", d.url)
```

Comment:
`devfile.yaml or .devfile.yaml not found in the provided url : %s`

Reply:
updated